pallets/storage-provider/src/lib.rs

```diff
@@ -388,6 +386,8 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(0)]
```

Comment:
You don't need to add the weight while you have `dev_mode`, remove this or use the right value.

Same applies for the rest of the file